tests/test-ajax.php

```diff
@@ -113,10 +139,39 @@ public function test_ajax_response_get_query_data_valid_query_with_filtered_colu
 		$this->assertTrue( $response->success );
 	}
 
+	/**
+	 * Test the AJAX response for fetching the database data with user capability.
+	 */
+	public function test_ajax_response_get_query_data_contributor_dissallow() {
+		wp_set_current_user( $this->contibutor_user_id );
+		$this->_setRole( 'contributor' );
+
+		$_GET['security'] = wp_create_nonce( Visualizer_Plugin::ACTION_FETCH_DB_DATA . Visualizer_Plugin::VERSION );
+
+		$_POST['params'] = array(
+			'query' => "/**/UPDATE wp_options SET option_value='administrator' WHERE option_name='default_role' --",
```

Comment:
From the new regex, I think this can be passed by using `INSERT INTO .... SLECT` or `UPDATE` with a subquery. I think we can account for this scenario. Example: https://regex101.com/r/8DsImW/1

Reply:
Thank you, this is a valid point. I improved the code and added a test for the case you described. 🚀